tests/rptest/tests/transactions_test.py

```diff
@@ -604,6 +605,107 @@ def consume_records(consumer, count):
 
         consume_records(consumer, 200)
 
+    @cluster(num_nodes=3)
+    def check_sequence_table_cleaning_after_eviction_test(self):
+        segment_size = 1024 * 1024
+        topic_spec = TopicSpec(partition_count=1, segment_bytes=segment_size)
+        topic = topic_spec.name
+
+        # make segments small
+        self.client().create_topic(topic_spec)
+
+        producers_cnt = 20
+
+        message_size = 128
+        segments_per_producer = 5
+        message_count = int(segments_per_producer * segment_size /
+                            message_size)
+        msg_body = random.randbytes(message_size)
+
+        producers = []
+        self.logger.info(f"producing {message_count} messages per producer")
+        for i in range(producers_cnt):
+            p = ck.Producer({
+                'bootstrap.servers': self.redpanda.brokers(),
+                'enable.idempotence': True,
+            })
+            producers.append(p)
+            for m in range(message_count):
+                p.produce(topic,
+                          str(f"p-{i}-{m}"),
+                          msg_body,
+                          partition=0,
+                          on_delivery=self.on_delivery)
+            p.flush()
+
+        def get_tx_metrics():
+            return self.redpanda.metrics_sample(
+                "tx_mem_tracker_consumption_bytes",
+                self.redpanda.started_nodes())
+
+        metrics = get_tx_metrics()
+        consumed_per_node = defaultdict(int)
+        for m in metrics.samples:
+            id = self.redpanda.node_id(m.node)
+            consumed_per_node[id] += int(m.value)
+        self.logger.info(
+            f"Bytes consumed by transactional subsystem: {consumed_per_node}")
+
+        self.client().alter_topic_config(
+            topic=topic, key=TopicSpec.PROPERTY_RETENTION_BYTES, value=128)
+        self.client().alter_topic_config(
+            topic=topic,
+            key=TopicSpec.PROPERTY_RETENTION_LOCAL_TARGET_BYTES,
+            value=128)
+
+        def segments_removed():
+            removed_per_node = defaultdict(int)
+            metric_sample = self.redpanda.metrics_sample(
+                "log_segments_removed", self.redpanda.started_nodes())
+            metric = metric_sample.label_filter(
+                dict(namespace="kafka", topic=topic))
+            for m in metric.samples:
+                removed_per_node[m.node] += m.value
+            return all([v > 0 for v in removed_per_node.values()])
+
+        wait_until(segments_removed, timeout_sec=60, backoff_sec=1)
+        # produce until next segment roll
+        #
+        # TODO: change this when we will implement cleanup on current,
+        # not the next eviction
+        last_producer = ck.Producer({
+            'bootstrap.servers':
+            self.redpanda.brokers(),
+            'enable.idempotence':
+            False,
+        })
+
+        for m in range(int(message_count / segments_per_producer) + 100):
```

Comment:
It isn't what's the meaning of `x=int(message_count / segments_per_producer) + 100`. Please add a comment of what it is (based on the definition of `message_count` it looks like `x` is the amount of messages enough to fill one segment) and why it's needed.